storage/innobase/dict/dict0stats.cc

```diff
@@ -1840,23 +1933,28 @@ dict_stats_scan_page(
 	return(offsets_rec);
 }
 
-/** Dive below the current position of a cursor and calculate the number of
-distinct records on the leaf page, when looking at the fist n_prefix
-columns. Also calculate the number of external pages pointed by records
-on the leaf page.
-@param[in]	cur			cursor
-@param[in]	n_prefix		look at the first n_prefix columns
-when comparing records
-@param[out]	n_diff			number of distinct records
-@param[out]	n_external_pages	number of external pages
-@return number of distinct records on the leaf page */
+/** Dive below the current position of a cursor and calculate
+the number of distinct records on the leaf page, when looking at
+the first n_prefix columns. Also calculate the number of external
+pages pointed by records on the leaf page.
+@param[in]	cur		cursor
+@param[in]	n_prefix	look at the first n_prefix columns
+				when comparing records
+@param[in]	stats		index scan statistics containing
+				nulls_unequal flag and
+				n_non_null_on_level pointer
+				for non-null counting
+@param[out]	leaf_stats	structure containing leaf page
+				statistics including
+				n_diff_on_leaf_page, n_external_pages
+				and n_non_null_on_leaf_page */
```

Comment:
There is no need to duplicate the definition of IndexScanStatistics and LeafPageStats in the comment. Instead, the comments should somehow point out the high-level difference between the two structures. What is it? (It is somewhat unclear to me as well.)

If `stats` is supposed to be input-only, why is the pointer not to `const`?